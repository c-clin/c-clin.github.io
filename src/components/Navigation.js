import React from 'react';
import { Link } from 'react-scroll';

const Navigation = () => {
  return (
    <div className="Navigation">
      <nav className="Navigation__container">
        <ul className="Navigation__list">
          <li className="Navigation__list-item">
            <Link
              to="Header"
              smooth={true}
              duration={500}
              className="list-item-link top-link-hover"
            >
              Home
            </Link>
          </li>
          <li className="Navigation__list-item">
            <Link
              to="About"
              smooth={true}
              duration={500}
              className="list-item-link top-link-hover"
            >
              About
            </Link>
          </li>
          <li className="Navigation__list-item">
            <Link
              to="Work"
              smooth={true}
              duration={600}
              className="list-item-link top-link-hover"
            >
              Work
            </Link>
          </li>
          <li className="Navigation__list-item">
            <Link
              to="Contact"
              smooth={true}
              duration={800}
              className="list-item-link top-link-hover"
            >
              Contact
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default Navigation;
