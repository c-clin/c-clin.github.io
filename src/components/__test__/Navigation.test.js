import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../Navigation';

describe('Navigation', () => {
  const navigation = shallow(<Navigation />);

  it('renders properly', () => {
    expect(navigation).toMatchSnapshot();
  });

  it('contains 4 links', () => {
    expect(navigation.find('.list-item-link')).toHaveLength(4);
  });
});
