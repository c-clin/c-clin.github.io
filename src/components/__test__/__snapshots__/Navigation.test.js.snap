// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="Navigation__container"
>
        <ul
                className="Navigation__list"
        >
                <li
                        className="Navigation__list-item"
                >
                        <Link
                                className="list-item-link top-link-hover"
                                duration={500}
                                offset={0}
                                smooth={true}
                                to="Header"
                        >
                                Home
                        </Link>
                </li>
                <li
                        className="Navigation__list-item"
                >
                        <Link
                                className="list-item-link top-link-hover"
                                duration={500}
                                offset={0}
                                smooth={true}
                                to="About"
                        >
                                About
                        </Link>
                </li>
                <li
                        className="Navigation__list-item"
                >
                        <Link
                                className="list-item-link top-link-hover"
                                duration={600}
                                offset={0}
                                smooth={true}
                                to="Work"
                        >
                                Work
                        </Link>
                </li>
                <li
                        className="Navigation__list-item"
                >
                        <Link
                                className="list-item-link top-link-hover"
                                duration={800}
                                offset={0}
                                smooth={true}
                                to="Contact"
                        >
                                Contact
                        </Link>
                </li>
        </ul>
</nav>,
      "className": "Navigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="Navigation__list"
>
          <li
                    className="Navigation__list-item"
          >
                    <Link
                              className="list-item-link top-link-hover"
                              duration={500}
                              offset={0}
                              smooth={true}
                              to="Header"
                    >
                              Home
                    </Link>
          </li>
          <li
                    className="Navigation__list-item"
          >
                    <Link
                              className="list-item-link top-link-hover"
                              duration={500}
                              offset={0}
                              smooth={true}
                              to="About"
                    >
                              About
                    </Link>
          </li>
          <li
                    className="Navigation__list-item"
          >
                    <Link
                              className="list-item-link top-link-hover"
                              duration={600}
                              offset={0}
                              smooth={true}
                              to="Work"
                    >
                              Work
                    </Link>
          </li>
          <li
                    className="Navigation__list-item"
          >
                    <Link
                              className="list-item-link top-link-hover"
                              duration={800}
                              offset={0}
                              smooth={true}
                              to="Contact"
                    >
                              Contact
                    </Link>
          </li>
</ul>,
        "className": "Navigation__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="Navigation__list-item"
>
              <Link
                            className="list-item-link top-link-hover"
                            duration={500}
                            offset={0}
                            smooth={true}
                            to="Header"
              >
                            Home
              </Link>
</li>,
            <li
              className="Navigation__list-item"
>
              <Link
                            className="list-item-link top-link-hover"
                            duration={500}
                            offset={0}
                            smooth={true}
                            to="About"
              >
                            About
              </Link>
</li>,
            <li
              className="Navigation__list-item"
>
              <Link
                            className="list-item-link top-link-hover"
                            duration={600}
                            offset={0}
                            smooth={true}
                            to="Work"
              >
                            Work
              </Link>
</li>,
            <li
              className="Navigation__list-item"
>
              <Link
                            className="list-item-link top-link-hover"
                            duration={800}
                            offset={0}
                            smooth={true}
                            to="Contact"
              >
                            Contact
              </Link>
</li>,
          ],
          "className": "Navigation__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="list-item-link top-link-hover"
                duration={500}
                offset={0}
                smooth={true}
                to="Header"
>
                Home
</Link>,
              "className": "Navigation__list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "className": "list-item-link top-link-hover",
                "duration": 500,
                "offset": 0,
                "smooth": true,
                "to": "Header",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="list-item-link top-link-hover"
                duration={500}
                offset={0}
                smooth={true}
                to="About"
>
                About
</Link>,
              "className": "Navigation__list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "About",
                "className": "list-item-link top-link-hover",
                "duration": 500,
                "offset": 0,
                "smooth": true,
                "to": "About",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="list-item-link top-link-hover"
                duration={600}
                offset={0}
                smooth={true}
                to="Work"
>
                Work
</Link>,
              "className": "Navigation__list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Work",
                "className": "list-item-link top-link-hover",
                "duration": 600,
                "offset": 0,
                "smooth": true,
                "to": "Work",
              },
              "ref": null,
              "rendered": "Work",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="list-item-link top-link-hover"
                duration={800}
                offset={0}
                smooth={true}
                to="Contact"
>
                Contact
</Link>,
              "className": "Navigation__list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Contact",
                "className": "list-item-link top-link-hover",
                "duration": 800,
                "offset": 0,
                "smooth": true,
                "to": "Contact",
              },
              "ref": null,
              "rendered": "Contact",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="Navigation__container"
>
          <ul
                    className="Navigation__list"
          >
                    <li
                              className="Navigation__list-item"
                    >
                              <Link
                                        className="list-item-link top-link-hover"
                                        duration={500}
                                        offset={0}
                                        smooth={true}
                                        to="Header"
                              >
                                        Home
                              </Link>
                    </li>
                    <li
                              className="Navigation__list-item"
                    >
                              <Link
                                        className="list-item-link top-link-hover"
                                        duration={500}
                                        offset={0}
                                        smooth={true}
                                        to="About"
                              >
                                        About
                              </Link>
                    </li>
                    <li
                              className="Navigation__list-item"
                    >
                              <Link
                                        className="list-item-link top-link-hover"
                                        duration={600}
                                        offset={0}
                                        smooth={true}
                                        to="Work"
                              >
                                        Work
                              </Link>
                    </li>
                    <li
                              className="Navigation__list-item"
                    >
                              <Link
                                        className="list-item-link top-link-hover"
                                        duration={800}
                                        offset={0}
                                        smooth={true}
                                        to="Contact"
                              >
                                        Contact
                              </Link>
                    </li>
          </ul>
</nav>,
        "className": "Navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="Navigation__list"
>
            <li
                        className="Navigation__list-item"
            >
                        <Link
                                    className="list-item-link top-link-hover"
                                    duration={500}
                                    offset={0}
                                    smooth={true}
                                    to="Header"
                        >
                                    Home
                        </Link>
            </li>
            <li
                        className="Navigation__list-item"
            >
                        <Link
                                    className="list-item-link top-link-hover"
                                    duration={500}
                                    offset={0}
                                    smooth={true}
                                    to="About"
                        >
                                    About
                        </Link>
            </li>
            <li
                        className="Navigation__list-item"
            >
                        <Link
                                    className="list-item-link top-link-hover"
                                    duration={600}
                                    offset={0}
                                    smooth={true}
                                    to="Work"
                        >
                                    Work
                        </Link>
            </li>
            <li
                        className="Navigation__list-item"
            >
                        <Link
                                    className="list-item-link top-link-hover"
                                    duration={800}
                                    offset={0}
                                    smooth={true}
                                    to="Contact"
                        >
                                    Contact
                        </Link>
            </li>
</ul>,
          "className": "Navigation__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="Navigation__list-item"
>
                <Link
                                className="list-item-link top-link-hover"
                                duration={500}
                                offset={0}
                                smooth={true}
                                to="Header"
                >
                                Home
                </Link>
</li>,
              <li
                className="Navigation__list-item"
>
                <Link
                                className="list-item-link top-link-hover"
                                duration={500}
                                offset={0}
                                smooth={true}
                                to="About"
                >
                                About
                </Link>
</li>,
              <li
                className="Navigation__list-item"
>
                <Link
                                className="list-item-link top-link-hover"
                                duration={600}
                                offset={0}
                                smooth={true}
                                to="Work"
                >
                                Work
                </Link>
</li>,
              <li
                className="Navigation__list-item"
>
                <Link
                                className="list-item-link top-link-hover"
                                duration={800}
                                offset={0}
                                smooth={true}
                                to="Contact"
                >
                                Contact
                </Link>
</li>,
            ],
            "className": "Navigation__list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="list-item-link top-link-hover"
                  duration={500}
                  offset={0}
                  smooth={true}
                  to="Header"
>
                  Home
</Link>,
                "className": "Navigation__list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": "list-item-link top-link-hover",
                  "duration": 500,
                  "offset": 0,
                  "smooth": true,
                  "to": "Header",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="list-item-link top-link-hover"
                  duration={500}
                  offset={0}
                  smooth={true}
                  to="About"
>
                  About
</Link>,
                "className": "Navigation__list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                  "className": "list-item-link top-link-hover",
                  "duration": 500,
                  "offset": 0,
                  "smooth": true,
                  "to": "About",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="list-item-link top-link-hover"
                  duration={600}
                  offset={0}
                  smooth={true}
                  to="Work"
>
                  Work
</Link>,
                "className": "Navigation__list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Work",
                  "className": "list-item-link top-link-hover",
                  "duration": 600,
                  "offset": 0,
                  "smooth": true,
                  "to": "Work",
                },
                "ref": null,
                "rendered": "Work",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="list-item-link top-link-hover"
                  duration={800}
                  offset={0}
                  smooth={true}
                  to="Contact"
>
                  Contact
</Link>,
                "className": "Navigation__list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Contact",
                  "className": "list-item-link top-link-hover",
                  "duration": 800,
                  "offset": 0,
                  "smooth": true,
                  "to": "Contact",
                },
                "ref": null,
                "rendered": "Contact",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
